scons/site_tools/hosted.py

```diff
@@ -101,5 +101,12 @@ def generate(env, **kw):
 		"-Woverloaded-virtual",
 	]
 
+	if 'CXXFLAGS' in os.environ:
+		flags = "".join(os.environ['CXXFLAGS'])
+		env['CXXFLAGS'].append(flags)
+	if 'LDFLAGS' in os.environ:
+		flags = "".join(os.environ['LDFLAGS'])
+		env['LINKFLAGS'] = flags
+
```

Comment:
Nah, I'm happy that it works now, so I'm not going to experiment further.
Build systems are not my thing ;-P